components/workspace-rollout-job/pkg/analysis/analyzer.go

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package analysis
+
+import (
+	"context"
+)
+
+type Analyzer interface {
+	// Given a cluster name, MoveForward is called by the rollout routine
+	// repeatedly to determine whether to move forward on the rollout or not.
+	// -1 means to revert
+	// 0 means that there is no data to make an informed decision
+	// 1 means to move forward based on data
```

Comment:
This can definitely be done in another PR (if you agree that it's a good idea), but what do you think about replacing those raw ints into a type?

Something like:

```go
type Decision int

const (
  DecisionRevert		Decision = -1
  DecisionMoveForward		Decision = 1
  DecisionInsuficientData	Decision = 0
)

type Analyzer interface {
	// Given a cluster name, MoveForward is called by the rollout routine
	// repeatedly to determine whether to move forward on the rollout or not.
	GetDecision(ctx context.Context, clusterName string) (Decision, error)
}
```

Everything would work exactly the same, the only difference is that the code will become more readable in `pkg/rollout/rollout.go`

```go
decision, err := r.analyzer.GetDecision(context.Background(), r.newCluster)

if decision == analyzer.DecisionRevert {
	r.revert <- true
	return
} else if decision == analyzer.DecisionInsuficientData && r.currentScore >= r.okayScoreUntilNoData  {
	r.revert <- true
	return
} 
log.Info("Analyzer says yes, rolling forward")

```
_(A bit of code was deleted just to make this comment more compact)_

Reply:
Definitely much better! Doing this for sure!

Reply:
Updated!